runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -0,0 +1,359 @@
+		{
+			return;
+		}
+
+		ChatPlayer player = getChatPlayerFromName(event.getMenuTarget());
+
+		if (player != null)
+		{
+			hop(player.getWorld());
+		}
+	}
+
+	void hop(boolean previous)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		int totalLevel = client.getTotalLevel();
+
+		World currentWorld = worldResult.findWorld(client.getWorld());
+
+		// Make it so you always hop out of PVP, high risk and bounty worlds
+		// Also don't regard skill total worlds as a type that must be hopped between
+		EnumSet<WorldType> currentWorldTypes = currentWorld.getTypes().clone();
+		currentWorldTypes.remove(WorldType.PVP);
+		currentWorldTypes.remove(WorldType.PVP_HIGH_RISK);
+		currentWorldTypes.remove(WorldType.BOUNTY);
+		currentWorldTypes.remove(WorldType.SKILL_TOTAL);
+
+		int currentWorldIndex = worldResult.getWorlds().indexOf(currentWorld);
+		CyclicListIterator<World> worldIterator = new CyclicListIterator<>(worldResult.getWorlds(), currentWorldIndex);
+
+		// Skip over current
+		if (!previous)
+		{
+			worldIterator.next();
+		}
+
+		World world;
+		do
+		{
+			if (previous)
+			{
+				world = worldIterator.previous();
+			}
+			else
+			{
+				world = worldIterator.next();
+			}
+
+			EnumSet<WorldType> types = world.getTypes().clone();
+
+			if (types.contains(WorldType.SKILL_TOTAL))
+			{
+				try
+				{
+					int totalRequirement = Integer.parseInt(world.getActivity().substring(0, world.getActivity().indexOf(" ")));
+
+					if (totalLevel >= totalRequirement)
+					{
+						types.remove(WorldType.SKILL_TOTAL);
+					}
+				}
+				catch (NumberFormatException ex)
+				{
+					log.error(null, ex);
+				}
+			}
+
+			if (currentWorldTypes.equals(types))
+			{
+				break;
+			}
+		}
+		while (world != currentWorld);
+
+		if (world == currentWorld)
+		{
+			String chatMessage = new ChatMessageBuilder()
+					.append(ChatColorType.NORMAL)
+					.append("Couldn't find a world to quick-hop to.")
+					.build();
+
+			chatMessageManager.queue(QueuedMessage.builder()
+					.type(ChatMessageType.GAME)
+					.runeLiteFormattedMessage(chatMessage)
+					.build());
+		}
+		else
+		{
+			hop(world.getId());
+		}
+	}
+
+	private void hop(int world)
+	{
+		// Don't try to hop if the world doesn't exist
+		if (worldResult.findWorld(world) == null)
+		{
+			return;
+		}
+
+		worldToHopTo = world;
+
+		String chatMessage = new ChatMessageBuilder()
+				.append(ChatColorType.NORMAL)
+				.append("Quick-hopping to World ")
+				.append(ChatColorType.HIGHLIGHT)
+				.append(worldToHopTo + "")
+				.append(ChatColorType.NORMAL)
+				.append("..")
+				.build();
+
+		chatMessageManager
+				.queue(QueuedMessage.builder()
+				.type(ChatMessageType.GAME)
+				.runeLiteFormattedMessage(chatMessage)
+				.build());
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (worldToHopTo == -1)
+		{
+			return;
+		}
+
+		if (client.getWidget(WidgetInfo.WORLD_SWITCHER_LIST) == null)
```

Comment:
I would not open the world hopper explicitly and simply require users to open it at least once

Reply:
Why?

Reply:
That is not the route to go if we are planning to have our own world hopper sidebar. Players will complain for days on end at the fact that you must open the in-game world hopper once for our hopper to work at all.

Reply:
Keep the behavior of other clients.

Reply:
Idk, i just feel like it is very-bot like and this is really pushing it.

Reply:
OSBuddy does exactly this, it's ok.